src/msw/thread.cpp

```diff
@@ -500,34 +504,50 @@ class wxThreadKeepAlive
 /* static */
 void wxThreadInternal::DoThreadOnExit(wxThread *thread)
 {
+    if ( wxSystemOptions::IsFalse("catch-unhandled-exceptions") )
+    {
+        thread->OnExit();
+    }
+    else
```

Comment:
And remove this one too.

Reply:
But then `thread->OnExit()` would get called twice.